### DISCUSSION TITLE
OPT-140-LOT difference between SDK and XSD

### DISCUSSION THREAD

**Discussion by ismailatitallah** [OP-TED/eForms-SDK#292] [Q&A] [answered]
Hello, I am generation a notice with type 16.
le field OPT-140-LOT is not mandatory in the description of the notice (16.json in SDK)
Schematron validation is OK. All rules passed.
However if OPT-140-LOT is not mentionned, The validation with the xsd does not pass even thought schematron is OK.

Is that a Bug?

**Comment by YvesJo** [comment, marked as answer]
Hi,
Schema and Schematron validations are complementary. Validation is a multi step process starting with lexical validation (XSD).
According to the XSD, as soon has you have /*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms/cac:CallForTendersDocumentReference in the XML instance, the element cbc:ID is required; when /*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms/cac:CallForTendersDocumentReference is not present, then its child cbc:ID is not allowed.
Presence of references to call  for tender documents is related to the presence of cac:CallForTendersDocumentReference (hence of cac:CallForTendersDocumentReference/cbc:ID in the /*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms context).
Procurement documents aren't always mandatory and some Conditionally Mandatory rules have been defined. Most of these rules are currently deactivated and will be activated progressively.
KR
